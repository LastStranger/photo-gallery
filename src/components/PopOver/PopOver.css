@import '../../common_styles/colors.css';

.popover {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(20, 20, 20, 1);
  justify-content: center;
  align-items: center;
  z-index: 10;
  opacity: 0;
  display: flex;
  transition: opacity 1s ease;

  &.is-visibile {
    transition: opacity 1s ease;
    opacity: 1;
  }
}

.popover-container {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  background-color: var(--info);
  color: white;
  min-width: 170px;
  min-height: 100px;
  border-radius: 5px;
}
