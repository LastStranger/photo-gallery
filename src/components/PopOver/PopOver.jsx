// @flow

import { h } from 'preact';
import './PopOver.css';

const PopOver = (props: { show: boolean }) =>
  <div className={`popover${props.show ? ' is-visibile' : ''}`} role="dialog" aria-labelledby="dialogTitle">
    <div className="popover-container">
      <h2 className="popover-title" id="dialogTitle">Loading</h2>
      <img className="popover-img" src="/assets/icons/loading.svg" alt="loading animation" />
    </div>
  </div>;

export default PopOver;
