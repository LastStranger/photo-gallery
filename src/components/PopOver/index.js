// @flow
import PopOver from './PopOver';

export default PopOver;
