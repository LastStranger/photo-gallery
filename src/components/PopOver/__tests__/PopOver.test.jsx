// @flow

import { h } from 'preact';
import render from 'preact-render-to-string';
import PopOver from '../';

test('renders with show', () => {
  const string = render(<PopOver show />);
  expect(string).toMatchSnapshot();
});

test('renders without show', () => {
  const string = render(<PopOver show={false} />);
  expect(string).toMatchSnapshot();
});
