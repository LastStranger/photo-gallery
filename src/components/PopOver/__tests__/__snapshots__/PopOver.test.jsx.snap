// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with show 1`] = `
<div class="popover is-visibile" role="dialog" aria-labelledby="dialogTitle">
  <div class="popover-container">
    <h2 class="popover-title" id="dialogTitle">Loading</h2>
    <img class="popover-img" src="/assets/icons/loading.svg" alt="loading animation"
    />
  </div>
</div>
`;

exports[`renders without show 1`] = `
<div class="popover" role="dialog" aria-labelledby="dialogTitle">
  <div class="popover-container">
    <h2 class="popover-title" id="dialogTitle">Loading</h2>
    <img class="popover-img" src="/assets/icons/loading.svg" alt="loading animation"
    />
  </div>
</div>
`;
